### PR TITLE
feat: add an RDS Module

### PR DESCRIPTION
**Pull request by CalvinRodo**
# Summary | Résumé

This is the first attemp no guarantees that it works

This creates an RDS proxy and a postgres RDS database, it will also
create security group and manage the proxy secret to connect the proxy
with the RDS cluster.

### CODE REVIEW

File: rds/input.tf [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+variable "sg_ids" {
+  description = "(required) The security groups this DB is to be attached to"
+  type        = set(string)
+}
+
+###
+# Database Admin User
+###
+
+variable "master_uname" {
```
Comment (maxneuvians) [inline rds/input.tf:75]: Suggest changing  to `username` to align with the discontinued use of the `master` noun

File: rds/input.tf [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  description = "(required) The username for the admin user for the db"
+  type        = string
+}
+
+variable "master_pword" {
```
Comment (maxneuvians) [inline rds/input.tf:80]: Suggest changing  to `password` to align with the discontinued use of the `master` noun

**Review by maxneuvians — APPROVED** [review]
Two suggestions otherwise looks good

File: rds/input.tf (hunk 1 above)
Comment (CalvinRodo) [reply rds/input.tf:75]: good idea I was just parroting the AWS terminology

File: rds/input.tf (hunk 2 above)
Comment (CalvinRodo) [reply rds/input.tf:80]: will do

**Pull request merged**